src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
 }
```

Comment:
This code is very difficult to read.
We can create a helper function for finding mothers with children.
```suggestion
function calculateWomenAverageAge(people, withChildren) {
  const listFemaleGender = people.filter(person => withChildren
    ? findWomenWithChildren(person, people)
    : person.sex === 'f'
  );

  let sum = 0;

  listFemaleGender.forEach((woman) => sum += woman.died - woman.born);

  return sum / listFemaleGender.length;
}

function findWomenWithChildren(person, arrPeople) {
  return ...;
}
```